Source/Plugins/Processing/Test/ConvertColorToGrayScaleTest.cpp

```diff
@@ -0,0 +1,547 @@
+  {
+    aa->initializeWithValue(0);
+    size_t index{0};
+    for(const auto& eachColor : testColors)
+    {
+      aa->setTuple(index++, eachColor);
+    }
+  }
+
+  static DataContainer::Pointer createVertexGeometryDataContainer(const DataArray<uint8_t>::Pointer& aa, const QVector<size_t>& tDims)
+  {
+    AttributeMatrix::Pointer am{AttributeMatrix::New(tDims, SIMPL::Defaults::VertexAttributeMatrixName, AttributeMatrix::Type::Vertex)};
+    am->addAttributeArray(aa->getName(), aa);
+
+    DataContainer::Pointer dc = DataContainer::New(SIMPL::Defaults::VertexDataContainerName);
+    dc->addAttributeMatrix(SIMPL::Defaults::VertexAttributeMatrixName, am);
+    dc->setGeometry(VertexGeom::CreateGeometry(static_cast<int64_t>(aa->getNumberOfTuples()), SIMPL::Geometry::VertexGeometry));
+    return dc;
+  }
+
+  int CheckFilterParameters(const QVariant& algorithm, const FloatVec3_t& cws, const uint8_t& cc)
+  {
+    QVariant conversionAlgorithm{m_colorToGrayscaleFilter->property("ConversionAlgorithm")};
+    FloatVec3_t colorWeights{(m_colorToGrayscaleFilter->property("ColorWeights").value<FloatVec3_t>())};
+    QVariant outputArrayPrefix{m_colorToGrayscaleFilter->property("OutputArrayPrefix")};
+    QVariant colorChannel{m_colorToGrayscaleFilter->property("ColorChannel")};
+    QVariant createNewAM{m_colorToGrayscaleFilter->property("CreateNewAttributeMatrix")};
+    QVariant outputAMName{m_colorToGrayscaleFilter->property("OutputAttributeMatrixName")};
+
+    int wrongParameters{0};
+    wrongParameters += (conversionAlgorithm == algorithm) ? 0 : 1;
+    float epsilon{0.000000001f};
+    wrongParameters += (abs(colorWeights.x - cws.x) < epsilon && abs(colorWeights.y - cws.y) < epsilon && abs(colorWeights.z - cws.z) < epsilon) ? 0 : 1;
+    wrongParameters += (outputArrayPrefix == m_outputArrayPrefix) ? 0 : 1;
+    wrongParameters += (colorChannel == cc) ? 0 : 1;
+    wrongParameters += (createNewAM == m_createNewAM) ? 0 : 1;
+    wrongParameters += (outputAMName == m_outputAMName) ? 0 : 1;
+
+    return wrongParameters;
+  }
+
+  int CompareResults(const uint8_t& algoMapIndex) const
+  {
+    QString amName{m_outputAMName};
+    DataContainer::Pointer dc{m_dca->getDataContainers().first()};
+    if(!m_createNewAM)
+      amName = dc->getAttributeMatrices().first()->getName();
+    AttributeMatrix::Pointer am{dc->getAttributeMatrix(amName)};
+
+    UCharArrayType::Pointer testArray{};
+    for(const auto& eachAAName : am->getAttributeArrayNames())
+    {
+      if(eachAAName.contains(m_outputArrayPrefix))
+        testArray = am->getAttributeArrayAs<UCharArrayType>(eachAAName);
+    }
+
+    std::vector<uint8_t> colorArray{algorithmMap[algoMapIndex]};
+
+    int wrongValues{0};
+    tbb::parallel_for(tbb::blocked_range<size_t>(0, colorArray.size()), [&](const tbb::blocked_range<size_t>& r) {
+      for(size_t index = r.begin(); index < r.end(); ++index)
+      {
+        uint8_t testValue{testArray->getValue(index)};
```

Comment:
uint8_t testValue = testArray->getValue(index);

Please use this style for initializing simple scalar values.